### PR TITLE
feat(ips): adds methods to HostbasedSetupService

### PR DESCRIPTION
**Pull request by Craig-Spencer-12**
## PR Checklist

- [x] Unit Tests have been added for new changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/amt/messages.ts [hunk 1]
```diff
@@ -144,6 +144,15 @@ export class Messages {
           body += `<r:LinkPolicy>${item}</r:LinkPolicy>`
         })
         body += `<r:MACAddress>${ethernetPortObject.MACAddress}</r:MACAddress><r:PhysicalConnectionType>${ethernetPortObject.PhysicalConnectionType}</r:PhysicalConnectionType><r:SharedDynamicIP>${String(ethernetPortObject.SharedDynamicIp)}</r:SharedDynamicIP><r:SharedMAC>${String(ethernetPortObject.SharedMAC)}</r:SharedMAC><r:SharedStaticIp>${String(ethernetPortObject.SharedStaticIp)}</r:SharedStaticIp></r:AMT_EthernetPortSettings></Body>`
+        // this.wsmanMessageCreator.createOtherBody('AMT_EthernetPortSettings', this.resourceUriBase, Classes.AMT_ETHERNET_PORT_SETTINGS, {
```
Comment (rsdmike) [inline src/amt/messages.ts:161]: can we remove this?

File: src/ips/actions.ts [hunk 2]
```diff
@@ -10,7 +10,9 @@ export enum Actions {
   PUT = 'http://schemas.xmlsoap.org/ws/2004/09/transfer/Put',
   DELETE = 'http://schemas.xmlsoap.org/ws/2004/09/transfer/Delete',
   SETUP = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_HostBasedSetupService/Setup',
+  ADMIN_SETUP = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_HostBasedSetupService/AdminSetupResponse',
+  ADD_NEXT_CERT_IN_CHAIN = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_HostBasedSetupService/AddNextCertInChainResponse',
   START_OPT_IN = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_OptInService/StartOptIn',
   CANCEL_OPT_IN = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_OptInService/CancelOptIn',
-  SEND_OPT_IN_CODE = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_OptInService/SendOptInCode'
+  SEND_OPT_IN_CODE = 'http://intel.com/wbem/wscim/1/ips-schema/1/IPS_OptInService/SendOptInCode',
```
Comment (rsdmike) [inline src/ips/actions.ts:17]: should this last line have a comma?

File: src/ips/messages.ts [hunk 3]
```diff
@@ -17,18 +17,18 @@ export class Messages {
...
-    const body: string = this.wsmanMessageCreator.createBody(Methods.GET)
+    const body: string = this.wsmanMessageCreator.createCommonBody(Methods.GET)
     return this.wsmanMessageCreator.createXml(header, body)
   }
 
   private readonly put = (action: AllActions, ipsClass: Classes, messageId: string, data: OptInServiceResponse): string => {
     const header: string = this.wsmanMessageCreator.createHeader(action, `${this.resourceUriBase}${ipsClass}`, messageId)
-    let body: string = this.wsmanMessageCreator.createPutBody(data)
-    body = body.replace(`<r:${Classes.IPS_OPT_IN_SERVICE}>`, `<r:IPS_OptInService xmlns:r="${this.resourceUriBase}${Classes.IPS_OPT_IN_SERVICE}">`)
+    const key = Object.keys(data)[0]
```
Comment (rsdmike) [inline src/ips/messages.ts:26]: is this guaranteed to always be the zero index? what if there are no keys? or multiple keys?

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/ips/actions.ts (hunk 2 above)
Comment (Craig-Spencer-12) [reply src/ips/actions.ts:17]: whoops golang likes that comma but I forgot that nobody else does

**Review by rsdmike — APPROVED** [review]

**Pull request merged**